### DISCUSSION TITLE
Model or Mesh with other primitives?

### DISCUSSION THREAD

**Discussion by mludlum** [StereoKit/StereoKit#1062] [Q&A]
I'd like to create models that have triangles, lines and text.  Am I understanding correctly that Mesh objects are pre-buffered and only support triangle primitives, while lines are sent to the graphics adapter every time in the run loop?   It seems I may run into a performance issue if I have a ton of lines and text that I'm drawing every frame.

I want to render a DXF file.  I have the parsing-into-primitives step covered.  In my case, they usually have lines and text, and sometimes faces.   I see that I can load wavefront file into a Model.  obj supports line elements in addition to faces.  So does that mean Models support lines under the hood and could be exposed in a new feature?   Would you recommend converting dxf to obj if what I'm looking for is not possible?

**Comment by maluoi** [comment]
In StereoKit, a Mesh is a collection of triangles, and a Model is a collection of Meshes and Materials with transforms. These don't include text or lines as part of their functionality, so when loading something like a Wavefront .obj, any text or line data would be ignored!

If you have a lot of lines that stay the same every frame, you could indeed benefit from a different line drawing system! SK's lines are meant to be more of a high-level solution to dynamic line rendering, and is best suited to debug visualization and a few other more dynamic uses. But it does still perform pretty well, so it's not a bad solution to at least get you off the ground until that level of extra performance is necessary.

I might recommend creating something like a DxfModel that extends Model functionality by also including your own support for text and lines.
